### PR TITLE
Implement version_conflict_policy to pin user's artifact versions

### LINKED ISSUE #107: User defined artifacts get overridden by transitive dependencies
If a user explicitly references a dependency, we should use that version and not ones brought through transitive dependencies. Basically, prioritize user defined dependencies over anything else. For example:

```
maven_install(
    artifacts = [
        "com.google.cloud:google-cloud-storage:1.66.0",
        "com.google.guava:guava:27.1-jre",
    ]
)
```

Should result in a dependency tree that contains `com.google.guava:guava:27.1-jre`, and not the one brought by `com.google.cloud:google-cloud-storage:1.66.0`.

### PR DESCRIPTION
**Pull request by beasleyr-vmw**
This commit implements `maven_install`'s `version_conflict_policy` attribute
as described by @jin's comments ([1] and [2]).

`version_conflict_policy = "pinned"` translates to Coursier's
`-V group:artifact:version` for all requested artifacts.  The default of
"default" is a no-op.

[1]: https://github.com/bazelbuild/rules_jvm_external/issues/107#issuecomment-481460441
[2]: https://github.com/bazelbuild/rules_jvm_external/issues/107#issuecomment-481884679

Resolves #107.

Testing Done:
- added new test case to //tests/unit/build_tests

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -152,6 +152,24 @@ pinned_maven_install()
...
+    ],
+    repositories = [
+        "https://repo1.maven.org/maven2",
+        "https://maven.google.com",
+    ],
+    version_conflict_policy = "pinned",
+    maven_install_json = "//:policy_pinned_testing_install.json",
+)
+
+load("@policy_pinned_testing//:defs.bzl", _policy_pinned_maven_install="pinned_maven_install")
```
Comment (jin) [inline WORKSPACE]:
```suggestion
load("@policy_pinned_testing//:defs.bzl", _policy_pinned_maven_install = "pinned_maven_install")
```

File: coursier.bzl [hunk 2]
```diff
@@ -638,6 +638,9 @@ def _coursier_fetch_impl(repository_ctx):
     cmd = _generate_coursier_command(repository_ctx)
     cmd.extend(["fetch"])
     cmd.extend(artifact_coordinates)
+    if repository_ctx.attr.version_conflict_policy == "pinned":
+        for coord in artifact_coordinates:
+            cmd.extend(["-V", coord])
```
Comment (jin) [inline coursier.bzl]:
```suggestion
            cmd.extend(["--force-version", coord])
```

File: coursier.bzl [hunk 3]
```diff
@@ -813,6 +816,15 @@ coursier_fetch = repository_rule(
         "use_unsafe_shared_cache": attr.bool(default = False),
         "excluded_artifacts": attr.string_list(default = []),  # list of artifacts to exclude
         "generate_compat_repositories": attr.bool(default = False),  # generate a compatible layer with repositories for each artifact
+        "version_conflict_policy": attr.string(
+            doc = """Policy for user-defined vs. transitive dependency version conflicts
+
+            If "pinned", choose the user's version unconditionally.
```
Comment (jin) [inline coursier.bzl]: "choose the user-specified version in maven_install unconditionally"?

**Comment by beasleyr-vmw** [comment]
> Thank you so much for doing this!
> 
> Could you please add a section in the README about this? Feel free to reuse the example in [#107 (comment)](https://github.com/bazelbuild/rules_jvm_external/issues/107#issuecomment-481884679).

Sure, np! README tweaked in latest fixup.

**Comment by beasleyr-vmw** [comment]
Just pushed one more fixup after discovering a problem w/ the implicit `,classifier=` from `utils.artifact_coordinate`. Lmk if that fixup is too hacky.

**Pull request merged**